211: Add companion toggle to zones

rpeterman-gp opened the pull request:
Closes #211 

### This PR...
- Adds slide toggle input for `isCompanion` prop to flight form.
- Adds `isCompanion` to Flight model.
- Adds `isCompanion` to the props that trigger an allocation preview.
- Updates flight-form spec.

rpeterman-gp committed:
feat(flight-form): add companion toggle to zones

sandikbarr reviewed (changes requested):
Can you make sure you're using the .prettier configuration in the project? I keep seeing these formatting changes especially to the templates in your PRs. I'm working on this same code, and it really makes merge conflicts more difficult.

rpeterman-gp commented:
I am setup to use the prettier config in the project. The template HTML files would actually not be formatted by prettier, but by the editor's HTML settings, which I do not think we are capturing in the committed .vscode settings. Looks like the unintended changes caused by my HTML formatting settings is the `html.format.wrapAttributes` setting. Mine is set at the user level to the default of `auto`, but it looks like that file had previously been formatted with `aligned-multiple`. Can we confirm that is the setting someone who has touch these files has been using, and if that is the setting that should be used. I can add it to the .vscode settings in the project.

rpeterman-gp committed:
config: update .vscode settings

- make prettier the default formatter
- make prettier the default formatter for html files

rpeterman-gp committed:
style(flight-form): run template through prettier

sandikbarr reviewed (approved):
Looks great! Thanks so much for sorting out that formatting, will be very helpful for comparing and merging this into my WIP zones ControlValueAccessor component.
